src/main/java/org/candlepin/dto/api/v1/SubscriptionTranslator.java

```diff
@@ -0,0 +1,116 @@
+        return this.translate(null, source);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SubscriptionDTO translate(ModelTranslator translator, Subscription source) {
+        return source != null ? this.populate(translator, source, new SubscriptionDTO()) : null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SubscriptionDTO populate(Subscription source, SubscriptionDTO destination) {
+        return this.populate(null, source, destination);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SubscriptionDTO populate(ModelTranslator translator, Subscription source, SubscriptionDTO dest) {
+        if (source == null) {
+            throw new IllegalArgumentException("source is null");
+        }
+
+        if (dest == null) {
+            throw new IllegalArgumentException("dest is null");
+        }
+
+        dest.id(source.getId())
+            .created(Util.toDateTime(source.getCreated()))
+            .updated(Util.toDateTime(source.getUpdated()))
+            .modified(Util.toDateTime(source.getModified()))
+            .lastModified(Util.toDateTime(source.getLastModified()))
+            .startDate(Util.toDateTime(source.getStartDate()))
+            .endDate(Util.toDateTime(source.getEndDate()))
+            .accountNumber(source.getAccountNumber())
+            .contractNumber(source.getContractNumber())
+            .orderNumber(source.getOrderNumber())
+            .quantity(source.getQuantity())
+            .stacked(source.isStacked())
+            .stackId(source.getStackId())
+            .upstreamConsumerId(source.getUpstreamConsumerId())
+            .upstreamEntitlementId(source.getUpstreamEntitlementId())
+            .upstreamPoolId(source.getUpstreamPoolId());
+
+        if (translator != null) {
+            dest.cdn(translator.translate(source.getCdn(), CdnDTO.class))
+                .certificate(translator.translate(source.getCertificate(), CertificateDTO.class))
+                .owner(translator.translate(source.getOwner(), NestedOwnerDTO.class))
+                .product(translator.translate(source.getProduct(), ProductDTO.class))
+                .derivedProduct(translator.translate(source.getDerivedProduct(), ProductDTO.class))
+                .providedProducts(translateProducts(translator, source.getProvidedProducts()))
+                .derivedProvidedProducts(translateProducts(translator, source.getDerivedProvidedProducts()));
+        }
+        else {
+            dest.providedProducts(new HashSet<>())
+                .derivedProvidedProducts(new HashSet<>());
+        }
+
+        return dest;
+    }
+
+    private Set<ProductDTO> translateProducts(ModelTranslator translator, Collection<ProductData> products) {
+        if (products == null) {
+            return new HashSet<>();
+        }
+
+        return products.stream()
+            .map(product -> translator.translate(product, ProductDTO.class))
```

Comment:
use stream mapper here

Reply:
Sorry. I forgot to delete this translator. Same with ProductDTO and ContentDTO translators. Done.